test/jdk/javax/swing/JList/4618767/JListSelectedElementTest.java

```diff
@@ -0,0 +1,217 @@
+    private static final int MENU = KeyEvent.VK_F;
+    private static JFrame frame;
+    private static JList<String> list;
+    private static Robot robot;
+    private static volatile boolean menuSelected;
+
+    public static void main(String[] args) throws Exception {
+        runTest();
+    }
+
+    public static void runTest() throws Exception {
+        robot = new Robot();
+        robot.setAutoWaitForIdle(true);
+        robot.setAutoDelay(200);
+
+        final boolean isAMac = System.getProperty("os.name").toLowerCase().contains("os x");
+
+        List<String> lafs = Arrays.stream(getInstalledLookAndFeels())
+                                  .map(UIManager.LookAndFeelInfo::getClassName)
+                                  .collect(Collectors.toList());
+        for (final String laf : lafs) {
+            try {
+                AtomicBoolean lafSetSuccess = new AtomicBoolean(false);
+                SwingUtilities.invokeAndWait(() -> {
+                    lafSetSuccess.set(setLookAndFeel(laf));
+                    if (lafSetSuccess.get()) {
+                        createUI();
+                    }
+                });
+                if (!lafSetSuccess.get()) {
+                    continue;
+                }
+                robot.waitForIdle();
+
+                // Request for focus and wait until the list has focus.
+                SwingUtilities.invokeAndWait(() -> list.requestFocusInWindow());
+                int waitCount = 0;
+                while (!isFocusOwner()) {
+                    robot.delay(100);
+                    waitCount++;
+                    if (waitCount > 20) {
+                        throw new RuntimeException("Waited for long, but can't get focus for list");
+                    }
+                }
+
+                // Select element named as 'bill'
+                robot.keyPress(KeyEvent.VK_B);
+                robot.keyRelease(KeyEvent.VK_B);
+
+                // Assertion check to verify that selected node is 'bill'
+                String elementSel = list.getSelectedValue();
+                if (!"bill".equals(elementSel)) {
+                    throw new RuntimeException("Test failed for " + laf
+                            + " as the list element selected: " + elementSel
+                            + " is not the expected one 'bill'"
+                    );
+                }
+
+                menuSelected = false;
+                // Now operate Menu using Mnemonics, different key combinations for different OS.
+                // For most of the OS its ALT+F, except non Nimbus LnFs in Mac, here its ALT+CNTRL+F.
+                if (isAMac && !laf.contains("Nimbus")) {
+                    robot.keyPress(KeyEvent.VK_ALT);
+                    robot.keyPress(KeyEvent.VK_CONTROL);
+                    robot.keyPress(MENU);
+                    robot.keyRelease(KeyEvent.VK_ALT);
+                    robot.keyRelease(KeyEvent.VK_CONTROL);
+                    robot.keyRelease(MENU);
```

Comment:
I suggest adding a helper method or using an existing one from reghelpers/Utils class:

```java
    public static void hitKeys(Robot robot, int... keys) {
        for (int i = 0; i < keys.length; i++) {
            robot.keyPress(keys[i]);
        }

        for (int i = keys.length - 1; i >= 0; i--) {
            robot.keyRelease(keys[i]);
        }
    }
```

If you copy the implementation, you can use your robot field instead of passing it as the parameter.

Also, it's a good practice to release in the reverse order to pressing them, which this method will help to achieve.